test/models/game_test.exs

```diff
@@ -1,18 +1,44 @@
 defmodule CoverMyPingPong.GameTest do
-  use CoverMyPingPong.ModelCase
+  use CoverMyPingPong.{ModelCase}
+  import CoverMyPingPong.Factory
 
-  alias CoverMyPingPong.Game
+  alias CoverMyPingPong.{Game}
 
-  @valid_attrs %{match_winner: "some content", opponent: "some content", opponent_score: 42, player: "some content", player_score: 42}
+  @valid_attrs %{
+    player_id: 1,
+    opponent_id: 2,
+    player_score: Enum.random(0..21),
+    opponent_score: Enum.random(0..21),
+    match_winner_id: 1,
+    played_at_date: "2017-06-10",
+    location: Enum.random(["Columbus", "Cleveland"])
+  }
   @invalid_attrs %{}
 
-  test "changeset with valid attributes" do
-    changeset = Game.changeset(%Game{}, @valid_attrs)
-    assert changeset.valid?
-  end
+  describe "changeset/2" do
+    test "with valid attributes is valid" do
+      changeset = Game.changeset(%Game{}, @valid_attrs)
+      assert changeset.valid?
+    end
+
+    test "is invalid without a player id" do
+      attrs = Map.delete(@valid_attrs, :player_id)
+      changeset = Game.changeset(%Game{}, attrs)
+      refute changeset.valid?
+    end
+
+    test "is invalid without an opponent id" do
+      attrs = Map.delete(@valid_attrs, :opponent_id)
+      changeset = Game.changeset(%Game{}, attrs)
+      refute changeset.valid?
+    end
 
-  test "changeset with invalid attributes" do
-    changeset = Game.changeset(%Game{}, @invalid_attrs)
-    refute changeset.valid?
+    Enum.each @valid_attrs, fn {attr, value} ->
```

Comment:
@Serneum Is this possible? I'd like to iterate over the `@valid_attrs` map and for each attribute create a new test. Currently, I am getting `undefined function attr/0`.  `IO.puts attr` outside the test..do works fine